### DISCUSSION TITLE
HammerDB on Linux+POWER

### DISCUSSION THREAD

**Discussion by jaopaulolc** [TPC-Council/HammerDB#442] [Q&A] [answered]
Hi,

It seems like HammerDB depends on `tclkit` which is only available for x86. Is it possible to run HammerDB on PowerPC?

**Comment by sm-shaw** [comment, marked as answer]
Hi, No HammerDB is not dependent on Tclkit ... yet. There are plans to simply the x86 install with tclkit/zipkit as described here https://github.com/TPC-Council/HammerDB/issues/383 however the current build that you download from the releases section will always be available as a development version as this includes all the source in text form. 

Also, there is full support to build from source with a single command as described here https://www.hammerdb.com/blog/uncategorized/how-to-build-hammerdb-from-source/. So the best starting point would be here to try and build for your platform.  The design approach has always been to ensure that all features support Linux and Windows equally and there are no current plans from the core team to support any platforms beyond x86 Linux and Windows with a discussion on this is here https://github.com/TPC-Council/HammerDB/discussions/282.  

As described there, if you would like to contribute the development and testing to make it work on another platform (and ongoing maintenance and support for all databases), we (TPC-OSS) would be happy to work with you on integrating such an effort.

**Reply by jaopaulolc** [reply]
Hi @sm-shaw .

Thanks for the prompt response.

I must be misunderstanding thing, because if I follow the step here, the first error that I get after running the `./Build-Linux.sh` script is:
`./Build-Linux.sh: line 46: ./tclkit-Linux64: cannot execute binary file: Exec format error`

Which makes sense because the `tlckit-Linux64` is the `tclkit` built for x86 and I am on a PowerPC machine. If `tclkit` is not available on PowerPC, how do I build HammerDB from source?

**Reply by sm-shaw** [reply]
OK so the tlckit-Linux64 is the Tcl environment that is used to bootstrap the build so it is not dependent on a system installed one. You would need another Tcl environment - however, as noted "there are no current plans from the core team to support any platforms beyond x86 Linux and Windows".
